front/pages/api/w/[wId]/members/me/assistant_visibility.ts

```diff
@@ -0,0 +1,192 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import {
+  deleteAgentVisibility,
+  upsertAgentVisibility,
+} from "@app/lib/api/assistant/visibility";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { Membership } from "@app/lib/models";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { MemberAgentVisibilityType } from "@app/types/assistant/agent";
+
+export type PostMemberAssistantVisibilityResponseBody = {
+  created: boolean | null;
+  visibility: MemberAgentVisibilityType;
+};
+
+export const PostMemberAssistantVisibilityRequestBodySchema = t.type({
+  assistantId: t.string,
+  visibility: t.union([
+    t.literal("workspace-unlisted"),
+    t.literal("published-listed"),
+  ]),
+});
+
+export const DeleteMemberAssistantsVisibilityBodySchema = t.type({
+  assistantId: t.string,
+});
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    PostMemberAssistantVisibilityResponseBody | { success: boolean }
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  const user = auth.user();
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_user_not_found",
+        message: "The user requested was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "POST":
+      const bodyValidation =
+        PostMemberAssistantVisibilityRequestBodySchema.decode(req.body);
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body: ${pathError}`,
+          },
+        });
+      }
+
+      const { assistantId, visibility } = bodyValidation.right;
+      const agentConfiguration = await getAgentConfiguration(auth, assistantId);
```

Comment:
I understood we wanted to double check taht (here & in the lib call)
So the agentconfig & membership check of ~20 lines is done 4 tiems
Tried to dry it for a while but didn't find a satisfactory solution